### PR TITLE
Update C++ kit with new configs and using removing "can_x" functions

### PR DESCRIPTION
**Pull request by StoneT2000**

### CODE REVIEW

**Comment by StoneT2000** [comment]
@themmj do you have a chance to take a look?

File: kits/cpp/src/lux/unit.cpp
Comment (themmj) [inline kits/cpp/src/lux/unit.cpp:56]: `canMove` still exists. In the python kit it was removed. Should it be removed here as well?
In general I'm not sure why those utilities are removed. Since the action queue cost lookup can be performed by the unit, I don't see why it shouldn't be able to make the call if it can move or not. Was there a particular reason for this?

File: kits/cpp/src/lux/unit.cpp
Comment (themmj) [inline kits/cpp/src/lux/unit.cpp:11]: The signature of this (`int64_t actionQueueCost(const Observation &obs) const;`) must be added to the `Unit` struct in `unit.hpp`.
Otherwise this will not compile.

File: kits/cpp/src/lux/config.hpp
Comment (themmj) [inline kits/cpp/src/lux/config.hpp:78]: This struct will not compile for multiple reasons:
a) The struct type inside the struct itself is a level of recursion that is not allowed :D
b) The implementation for the operator overload is missing.

In case [this](https://github.com/Lux-AI-Challenge/Lux-Design-2022/blob/dde666dc9c012d263623e08ecef82516f929d8c2/kits/sample_env_cfg.json#L59) is an up to date version of what we can expect `UNIT_ACTION_QUEUE_POWER_COST` to look like the struct should be:
```cpp
    struct ActionQueueConfigs {
        int64_t HEAVY;
        int64_t LIGHT;
        int64_t operator[](const std::string &name) const;
    };
    NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(ActionQueueConfigs, HEAVY, LIGHT)
```
This addresses a).
To address b) you can add the following to `config.cpp`
```cpp
    int64_t ActionQueueConfigs::operator[](const std::string &name) const {
        if (name == "HEAVY") {
            return HEAVY;
        }
        return LIGHT;
    }
```

**Review by themmj — CHANGES_REQUESTED** [review]
Hi @StoneT2000, there are a few things that need to be fixed. The major issue is the new struct around move queue configs. I can implement these changes if you want to. Just let me know.
Besides that I have a question around canX utilities. See my comment there.

File: kits/cpp/src/lux/unit.cpp
Comment (StoneT2000) [reply kits/cpp/src/lux/unit.cpp:56]: Yes remove. The reasoning is submitting an action queue it costs power. So canMove etc don't quite make sense since the action queue costa additional power.

**Comment by StoneT2000** [comment]
Yeah can you help add the fixes? I wrote something very quickly to convey the idea.

**Comment by themmj** [comment]
Maybe it's too late for this question. But wouldn't it be more intuitive if the action queue power cost was part of the unit config instead of a separate config entry which differentiates between LIGHT and HEAVY?

**Comment by themmj** [comment]
On another note, I don't have push rights to this repo. Should I make a PR from my fork to this branch instead?

**Comment by StoneT2000** [comment]
> Maybe it's too late for this question. But wouldn't it be more intuitive if the action queue power cost was part of the unit config instead of a separate config entry which differentiates between LIGHT and HEAVY?

yeah hindsight this is probably better but I think we can't make any updates to the engine now due to beta being close to release and the servers having rolled out already. For official release we can do this.

**Comment by StoneT2000** [comment]
@themmj yes a fork and PR works too. I'll close this once your PR is merged.

**Pull request merged**
